lib/logstash/filters/mutate.rb

```diff
@@ -202,37 +211,36 @@ def register
         :replacement  => replacement
       }
     end
-  end # def register
 
-  public
-  def filter(event)
-    
-
-    rename(event) if @rename
-    update(event) if @update
-    replace(event) if @replace
-    convert(event) if @convert
-    gsub(event) if @gsub
-    uppercase(event) if @uppercase
-    lowercase(event) if @lowercase
-    strip(event) if @strip
-    remove(event) if @remove
-    split(event) if @split
-    join(event) if @join
-    merge(event) if @merge
+    @mutators = []
```

Comment:
since most configs have 1 or 2 mutations, I figured that doing 1 or 2 loop iteration would more efficient than doing a systematic 12 `if` evaluation. I just wrote a [simple bench](https://gist.github.com/colinsurprenant/5ce33931a0c4ef662a16)  and `if` are in fact a tad faster so basically its really a wash. all-in-all even if it was twice faster or slower it has not significant performance impact because these cycles are orders of magnitude smaller than the computations done in the mutators themselves. 

I know that loop unrolling will always be faster but here my thinking was that the loop was going to be significantly smaller that  the number of `if` evaluation. 

@purbon it does not change nothing, its just another way of calling the mutators at each filter call.

@guyboertje don't know but I think calling a method if more costly than evaluating an ivar? 

anyway, personally I find my refactor more clear and the filter method cleaner but I really don't care here. both are "correct"